include/ocpp/v201/database_handler.hpp

```diff
@@ -31,21 +31,18 @@ class DatabaseHandler : public ocpp::common::DatabaseHandlerBase {
     template <typename T>
     void init_enum_table(const std::string& table_name, T begin, T end, std::function<std::string(T)> conversion);
 
-    bool clear_table(const std::string& table_name);
-
     // Availability management (internal helpers)
     // Setting evse_id to 0 addresses the whole CS, setting evse_id > 0 and connector_id=0 addresses a whole EVSE
     void insert_availability(int32_t evse_id, int32_t connector_id, OperationalStatusEnum operational_status,
                              bool replace);
     OperationalStatusEnum get_availability(int32_t evse_id, int32_t connector_id);
 
 public:
-    DatabaseHandler(const fs::path& database_path, const fs::path& sql_init_path);
+    DatabaseHandler(std::shared_ptr<common::DatabaseConnectionInterface> database, const fs::path& sql_init_path);
 
     /// \brief Opens connection to database file
     void open_connection();
```

Comment:
`open_connection` and `close_connection` could be moved to the `DatabaseHandlerCommon`. This requires some additional initialization adjustments because of the `nr_of_connectors` argument for `open_connection` in v16. We could also do this in another PR. What do you think?

Reply:
I see a relatively easy way of doing this so will do it in this PR :+1: